### DISCUSSION TITLE
canopy spread issues

### DISCUSSION THREAD

**Discussion by ckoven** [NGEET/fates#990] [General]
A thing occurred to me recently that I am not sure about how we should deal with. Since https://github.com/NGEET/fates/pull/279 the canopy spread logic has been at the site rather than the patch level. In the case of a nocomp configuration, if the total area in which woody PFTs can grow is less than the [canopy closure threshold parameter](https://github.com/NGEET/fates/blob/main/parameter_files/fates_params_default.cdl#L1406) on a given gridcell, then the trees will always remain in the high-crown-area state.  This doesn't seem ideal to me but I am not sure what the best alternative is. We could change the canopy spread parameter when in nocomp configuration to make it a vector of length nocomp PFTs and evaluate it separately on patches of each nocomp PFT. Or maybe we could leave as-is. Curious if others see this as a problem in the current scheme or if anyone sees a better solution.

**Comment by ckoven** [comment]
Also noting that this will also apply when and if we have managed land-use types that exclude woody plants: if their total area approaches one minus the canopy closure threshold, then that would change the crown area allometry of trees in the unmanaged land-use types.

**Comment by ckoven** [comment]
We discussed this today and noted that another potential solution to this would be to set our default values of `fates_allom_d2ca_coefficient_min` and `fates_allom_d2ca_coefficient_max` to have the same value, as part of our global calibration exercise using crown area allometries from TALLO or BAAD or similar.  This would both simplify the process of calibrating the crown area allometries as well as avoid issues like the ones I mention above in this thread, as well as others (e.g. as was encountered in https://github.com/NGEET/fates/issues/714).

**Reply by rosiealice** [reply]
Just thinking about this,  and maybe this is obvious, but in the absence of this change, the behaviour of nocomp mode will be both substantially different to fully dynamic mode, and quasi randomly affected by the pft timing. So perhaps it is a higher priority than we might imagine for anyone using this capability?
